### PR TITLE
Publish the k0s_sort binary

### PR DESCRIPTION
**Pull request by kke**
Adds a couple of features to the `k0s_sort` binary and publishes it as release assets.

```console
Usage: k0s_sort [options] [filename ...]
  -l	only print the latest version from input
  -o	print the latest version from online
  -s	omit prerelease versions
  -v	print k0s_sort version
```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    
+      - name: Tag name
+        id: tag-name
+        run: echo "tag=${GITHUB_REF#refs/tags/}" >> "$GITHUB_OUTPUT"
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+          check-latest: true
```
Comment (twz123) [inline .github/workflows/release.yml:24]: Hmm, this means the go version in the release pipeline might be different from the one being used when building the pull requests.

File: cmd/k0s_sort/k0s_sort.go
Comment (twz123) [inline cmd/k0s_sort/k0s_sort.go:1]: Any reason for the rename? I think it makes sense to have a main.go that contains the main function. I'm always a bit hesitant to use underscores in go file names, as this is also a mechanism used for conditional compilation.

**Review by twz123 — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+$(bins):
+	$(eval temp := $(subst -, ,$(subst $(BIN_PREFIX),,$(notdir $@))))
+	$(eval OS := $(word 1, $(subst -, ,$(temp))))
+	$(eval ARCH := $(word 2, $(subst -, ,$(temp))))
+	GOOS=$(OS) GOARCH=$(ARCH) CGO_ENABLED=0 go build $(BUILD_FLAGS) -o $@ ./cmd/k0s_sort
+
+bin/sha256sums.txt: $(bins)
+	sha256sum -b $(bins) | sed 's|bin/||' > $@
+
+build-all: $(bins) bin/sha256sums.md
```
Comment (twz123) [inline Makefile:21]: Oh, that's a typo

```suggestion
build-all: $(bins) bin/sha256sums.txt
```

File: .github/workflows/release.yml (hunk 1 above)
Comment (kke) [reply .github/workflows/release.yml]: Copy-paste

**Pull request merged**